api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -76,6 +76,22 @@
    */
   String TARGET_FILE_SIZE_BYTES = "target-file-size-bytes";
 
+  /**
+   * Determines if the data rewrite action should also remove non-global deletes associated with the data files.
+   * By enabling this option, any data filter specified through {@link #filter(Expression)} will be converted to
+   * an inclusive partition filter based on all the historical partition specs of the table.
```

Comment:
My understanding is that for non-global deletes, as long as the the filter is a partition filter, if we compact all the data files produced by the plan, the delete file can be safely removed. This is inefficient because technically we should do the following:
1. get all data files satisfying the filter
2. get the delete files of the data files
3. for the delete files, find the connected component (if we view this as a dependency graph of files), which might produce a much smaller subset of data files to compact
4. replan tasks based on the set of data and delete files

But that goes too far away from the RewriteDataFiles action, and might be an overkill in the end.

Reply:
Maybe we are saying the same thing, but why cant the action be to find the set of data files which are most impacted by delete files (i.e. have the most delete files corresponding to them); and then just merge the delete files into these data files , producing new data files. It seems to me just like the size based strategy has file size based thresholds, the merge of delete files strategy should have thresholds based on number of delete files corresponding to the data files.